pkg/k8s/errors.go

```diff
@@ -233,3 +233,17 @@ func (e *ImmutableFieldsMutationError) Error() string {
 func NewImmutableFieldsMutationError(immutableFields []string) *ImmutableFieldsMutationError {
 	return &ImmutableFieldsMutationError{immutableFields}
 }
+
+type UpdateNotSupportedError struct {
+	resourceGVK    schema.GroupVersionKind
```

Comment:
nit: GVK may not be appropriate (maybe GK?), but it really refers to how we want to handle this error in the reconcile. Suggest adding this error in k8s only when we plan to handle this error.